plugins/lsp/language-servers/rust-analyzer-config.nix

```diff
@@ -1,167 +1,65 @@
-        types.addCheck types.number (x: x <= maximum)
+        lib.types.addCheck lib.types.number (x: x <= maximum)
       else
-        types.number
-    else if type == "integer" then
+        lib.types.number
+    else if kind == "integer" then
+      let
+        inherit (typeInfo) minimum maximum;
+      in
       if minimum != null && maximum != null then
-        types.ints.between minimum maximum
+        lib.types.ints.between minimum maximum
       else if minimum != null then
-        types.addCheck types.int (x: x >= minimum)
+        lib.types.addCheck lib.types.int (x: x >= minimum)
       else if maximum != null then
-        types.addCheck types.int (x: x <= maximum)
+        lib.types.addCheck lib.types.int (x: x <= maximum)
       else
-        types.int
+        lib.types.int
+    else if kind == "object" then
+      lib.types.attrsOf lib.types.anything
+    else if kind == "string" then
+      lib.types.str
+    else if kind == "boolean" then
+      lib.types.bool
     else
-      throw "unhandled type `${builtins.toJSON type}` (in ${opt})";
+      throw "Unknown type: ${kind}";
 
-  mkRustAnalyzerOption =
-    opt:
-    {
-      default,
-      markdownDescription,
-      type ? null,
-      anyOf ? null,
-      # Enum types
-      enum ? null,
-      enumDescriptions ? null,
-      # Int types
-      minimum ? null,
-      maximum ? null,
-      # List types
-      items ? null,
-      uniqueItems ? false,
-    }:
-    mkOption {
-      type = types.nullOr (
-        mkRustAnalyzerType opt {
-          inherit
-            type
-            enum
-            minimum
-            maximum
-            items
-            anyOf
-            uniqueItems
-            ;
-        }
-      );
+  mkNixOption =
+    { description, type }:
+    lib.mkOption {
       default = null;
-      description =
-        if enum != null then
-          let
-            valueDesc = builtins.map ({ fst, snd }: ''- ${fst}: ${snd}'') (
-              lists.zipLists enum enumDescriptions
-            );
-          in
-          ''
-            ${markdownDescription}
-
-            Values:
-            ${builtins.concatStringsSep "\n" valueDesc}
-
-            ```nix
-            ${generators.toPretty { } default}
-            ```
-          ''
-        else
-          ''
-            ${markdownDescription}
-
-            default:
-            ```nix
-            ${generators.toPretty { } default}
-            ```
-          '';
+      inherit description;
+      type = lib.types.nullOr (mkRustAnalyzerType type);
     };
 
-  nixOptions = builtins.mapAttrs mkRustAnalyzerOption rustAnalyzerOptions;
-
   nestOpt =
     opt: value:
     let
-      parts = strings.splitString "." opt;
+      parts = lib.strings.splitString "." opt;
     in
-    builtins.foldl' (current: segment: { ${segment} = current; }) value (lists.reverseList parts);
+    builtins.foldl' (current: segment: { ${segment} = current; }) value (lib.lists.reverseList parts);
```

Comment:
Two observations:

- You could use `foldr` instead of reversing the list
- This is just a re-implementation of [`lib.setAttrByPath`](https://noogle.dev/f/lib/setAttrByPath)